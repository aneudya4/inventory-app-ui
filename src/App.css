@import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,400;1,300;1,400&display=swap');

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: 'Montserrat', sans-serif;
}

body {
  background: whitesmoke;
}
ul {
  list-style-type: none;
}
a,
a:active,
a:visited {
  text-decoration: none;
  color: white;
}

.btn {
  padding: 10px 0;
  border: 0;
  color: white;
  font-size: 15px;
  background: #82c9c1;
  cursor: pointer;
}
