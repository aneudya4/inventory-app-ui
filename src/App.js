import './App.css';
import SignUpWrapper from './components/signUpWrapper/SignUpWrapper';
import Header from './components/header/Header';
import Dashboard from './components/dashboard/Dashboard';
import { Route } from 'react-router-dom';
function App() {
  return (
    <div className='App'>
      <Route
        exact
        path={['/', '/accounts/register', '/accounts/login']}
        component={Header}
      />
      <Route path='/auth/dashboard' component={Dashboard} />
      <Route path='/accounts' component={SignUpWrapper} />
    </div>
  );
}

export default App;
