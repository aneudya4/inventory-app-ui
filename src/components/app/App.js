import React, { useState, useEffect } from 'react';
import './App.css';
import AuthFormsWrapper from '../authFormsWrapper/AuthFormsWrapper';
import Header from '../header/Header';
import Dashboard from '../dashboard/Dashboard';
import Homepage from '../homepage/Homepage';
import ApiContext from '../../apiContext';
import ServerError from '../server-error/ServerError';
import NotFound from '../not-found/NotFound';
import app from '../../firebaseConfig';
import { Route, Switch } from 'react-router-dom';
function App() {
  const [currentUser, setCurrentUser] = useState(null);

  useEffect(() => {
    app.auth().onAuthStateChanged(setCurrentUser);
  }, []);

  const value = {
    user: currentUser,
    setUser: setCurrentUser,
  };

  return (
    <div className='App'>
      <ApiContext.Provider value={value}>
        <Route
          exact
          path={['/', '/accounts/register', '/accounts/login']}
          component={Header}
        />
        <Switch>
          <Route path='/auth/dashboard' component={Dashboard} />
          <Route path='/accounts' component={AuthFormsWrapper} />
          <Route exact path='/server-error' component={ServerError} />
          <Route exact path='/' component={Homepage} />
          <Route path='*' component={NotFound} />
        </Switch>
      </ApiContext.Provider>
    </div>
  );
}

export default App;
