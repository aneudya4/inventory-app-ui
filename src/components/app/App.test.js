import { shallow } from 'enzyme';

import App from './App';

describe('Test on the <App/> Component', () => {
  it('renders App without crashing', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
    // render(<App />);
    // const linkElement = screen.getByText(/learn react/i);
    // expect(linkElement).toBeInTheDocument();
  });
});
