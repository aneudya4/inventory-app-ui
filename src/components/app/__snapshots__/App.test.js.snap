// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <App/> Component renders App without crashing 1`] = `
<div
  className="App"
>
  <ContextProvider
    value={
      Object {
        "setUser": [Function],
        "user": null,
      }
    }
  >
    <Route
      component={[Function]}
      exact={true}
      path={
        Array [
          "/",
          "/accounts/register",
          "/accounts/login",
        ]
      }
    />
    <Switch>
      <Route
        component={[Function]}
        path="/auth/dashboard"
      />
      <Route
        component={[Function]}
        path="/accounts"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/server-error"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="*"
      />
    </Switch>
  </ContextProvider>
</div>
`;
